src/mapengine/leaflet/js/typeLibrary/leaflet-eng.typeLibrary.feature.milstd.js

```diff
@@ -133,125 +145,169 @@ leafLet.internalPrivateClass.MilStdFeature = function() {
+          }
+
+          for (sModifier in oMainModifiers) {
+            if (oMainModifiers.hasOwnProperty(sModifier)) {
+              switch (sModifier) {
+                case leafLet.utils.milstd.longModifiers.FILL_COLOR:
+                  if (L.Browser.canvas) {
+                    oModifiers[msa.FillColor] = '#' + oMainModifiers[sModifier];
+                  } else {
+                    oModifiers['fillcolor'] = oMainModifiers[sModifier];
+                  }
+                  break;
+                case leafLet.utils.milstd.longModifiers.LINE_COLOR:
+                  if (L.Browser.canvas) {
+                    oModifiers[msa.LineColor] = '#' + oMainModifiers[sModifier];
+                  } else {
+                    oModifiers['linecolor'] = oMainModifiers[sModifier];
+                  }
+                  break;
+                case leafLet.utils.milstd.Modifiers.STANDARD:
+                  oModifiers[sModifier] = oMainModifiers[sModifier];
+                  break;
+                default:
+                  if (this._modifierOnList(sModifier)) {
+                    oModifiers[sModifier] = oMainModifiers[sModifier];
+                  }
+                  break;
+              }
             }
-
-            for (sModifier in oMainModifiers) {
-                if (oMainModifiers.hasOwnProperty(sModifier)) {
-                    switch (sModifier) {
-                    case leafLet.utils.milstd.longModifiers.FILL_COLOR:
-                        if (L.Browser.canvas) {
-                            oModifiers[msa.FillColor] = '#' + oMainModifiers[sModifier];
-                        } else {
-                            oModifiers['fillcolor'] = oMainModifiers[sModifier];
-                        }
-                        break;
-                    case leafLet.utils.milstd.longModifiers.LINE_COLOR:
-                        if (L.Browser.canvas) {
-                            oModifiers[msa.LineColor] = '#' + oMainModifiers[sModifier];
-                        } else {
-                            oModifiers['linecolor'] = oMainModifiers[sModifier];
-                        }
-                        break;
-                    case leafLet.utils.milstd.Modifiers.STANDARD:
-                        oModifiers[sModifier] = oMainModifiers[sModifier];
-                        break;
+          }
+
+          oModifiers.SIZE = this.getEngineInstanceInterface().iMilStdIconSize;
+          if (this.isSelected()) {
+            sClassName = 'icon-selected';
+          }
+
+          if (L.Browser.canvas) {
+            image = armyc2.c2sd.renderer.MilStdIconRenderer.Render(oItem.data.symbolCode, oModifiers);
+            iconUrl = image.toDataUrl();
+            oOffset = image.getCenterPoint();
+            oImageBounds = image.getImageBounds();
+            renderingOptimization = this.getEngineInstanceInterface().renderingOptimization;
+            oMilStdIcon;
+            if (renderingOptimization.viewInZone) {
```

Comment:
There are more affiliations than just these 4, see MIL-STD-2525C.   They all boil down to these 3 colors though.